### ISSUE TITLE
Platforms for Map2DFusion

### ISSUE THREAD

**Issue by MartinPaques** [zdzhaoyong/Map2DFusion#8]
Hi Yong,

I'm contacting you to talk about your algorithm Map2DFusion. My name is Martin Paques and i'm an French engineering student. For our last year project, my team and I plan to create a fixed wing drone to map post disaster areas. For this, we want to embed a jetson TX2 and run Map2DFusion on it. But I have some questions about it:
First, how do I have to modify your code to make it run on real time data (video stream from a GoPro and GPS data from our pixhawk mini) and not on your dataset ? 
Secondly, you said on the read me that to enable CUDA we just have to define the CUDA_PATH, but it's not trivial for me, would you mind to explain how I can do it ? 
My last question is just pure curiosity, have you ever run Map2DFusion embedded ? And what do you think about running it on a Jetson TX2 (maybe I will have to modify the FPS a bit) ? Have you ever think about running Map2D Fusion on a raspberry pi at 1 Hz ? 

Thanks in advance,

Ps: Your work is awesome.

Martin PAQUES
ESTACA Student 
Aerospace Major

**Comment by gpldecha** [comment]
@MartinPaques Have you had any success ?

**Comment by MartinPaques** [comment]
@gpldecha Yep i'm planning to release a Raspberry version within the next weeks

**Comment by fredlllll** [comment]
@MartinPaques im trying to run image sets with just gps data with this, but i dont know how to create the trajectory.txt. how are you getting the quaternion rotation from your images?

with defining the cuda path i guess he meant that you have to set a environment variable CUDA_PATH to the location of CUDA on linux. i think this is how you install it http://developer.download.nvidia.com/compute/cuda/7.5/Prod/docs/sidebar/CUDA_Installation_Guide_Linux.pdf

**Comment by NeXT1995** [comment]
@fredlllll How did you get trajectory.txt later? What about the effect

**Comment by fredlllll** [comment]
@NeXT1995 what? are you sure you read my post correctly?

**Comment by NeXT1995** [comment]
@MartinPaques  How do you get the trajectory.txt file?  And the Seven parameters in config.cfg？

**Comment by immuhammadadil** [comment]
Is it necessary to generate trajectory file using SLAM or can we use MATLAB also?

**Comment by MathsionYang** [comment]
> Hi Yong,
> 
> I'm contacting you to talk about your algorithm Map2DFusion. My name is Martin Paques and i'm an French engineering student. For our last year project, my team and I plan to create a fixed wing drone to map post disaster areas. For this, we want to embed a jetson TX2 and run Map2DFusion on it. But I have some questions about it:
> First, how do I have to modify your code to make it run on real time data (video stream from a GoPro and GPS data from our pixhawk mini) and not on your dataset ?
> Secondly, you said on the read me that to enable CUDA we just have to define the CUDA_PATH, but it's not trivial for me, would you mind to explain how I can do it ?
> My last question is just pure curiosity, have you ever run Map2DFusion embedded ? And what do you think about running it on a Jetson TX2 (maybe I will have to modify the FPS a bit) ? Have you ever think about running Map2D Fusion on a raspberry pi at 1 Hz ?
> 
> Thanks in advance,
> 
> Ps: Your work is awesome.
> 
> Martin PAQUES
> ESTACA Student
> Aerospace Major



> 是否有必要使用SLAM生成轨迹文件，还是可以使用MATLAB？

请问你有运行成功吗

**Comment by deepConnectionism** [comment]
Warnning: Compilation with CUDA can be enabled after CUDA_PATH defined.

How to use CUDA?

**Comment by deepConnectionism** [comment]
I already solved the CUDA problem. 
I am currently migrating to the jetson xavier nx platform. Installing QT4 is a challenge.
I have finished running this code in cuda mode for jetson xavier nx.
